pydbgen/pydbgen.py

```diff
@@ -45,266 +69,151 @@ def license_plate(self,seed=None,style=None):
-            
-            # Personal, official email
-            if data_type=='email':
-                for _ in range(num):
-                    lst.append(fake.email())
-                return pd.Series(lst)
-            if data_type=='office_email':
-                for _ in range(num):
-                    lst.append(fake.company_email())
-                return pd.Series(lst)
-            
-            # Company, Job title
-            if data_type=='company':
-                for _ in range(num):
-                    lst.append(fake.company())
-                return pd.Series(lst)
-            if data_type=='job_title':
-                for _ in range(num):
-                    lst.append(fake.job())
-                return pd.Series(lst)
-            
-            # Phone number, license plate (3 styles)
-            if data_type=='phone_number_simple':
-                for _ in range(num):
-                    lst.append(self.simple_ph_num())
-                return pd.Series(lst)
-            if data_type=='phone_number_full':
-                for _ in range(num):
-                    lst.append(fake.phone_number())
-                return pd.Series(lst)
-            if data_type=='license_plate':
-                for _ in range(num):
-                    lst.append(self.license_plate())
-                return pd.Series(lst)
-
-    def gen_dataframe(self,num=10,fields=['name'], real_email=True, real_city=True, phone_simple=True, seed=None):
+            raise ValueError('Number of samples must be a positive integer, found ' + num)
+
+        if num <= 0:
+            raise ValueError('Number of samples must be a positive integer, found ' + num)
+
+        num = int(num)
+        fake = self.fake
+        fake.seed(self.seed)
+
+        func_lookup = {
+            'name': fake.name,
+            'country': fake.country,
+            'street_address': fake.street_address,
+            'city': fake.city,
+            'real_city': self.city_real,
+            'state': fake.state,
+            'zipcode': fake.zipcode,
+            'latitude': fake.latitude,
+            'longitude': fake.longitude,
+            'name_month': fake.month_name,
+            'weekday': fake.day_of_week,
+            'year': fake.year,
+            'time': fake.time,
+            'date': fake.date,
+            'ssn': fake.ssn,
+            'email': fake.email,
+            'office_email': fake.company_email,
+            'company': fake.company,
+            'job_title': fake.job,
+            'phone_number_simple': self.simple_ph_num,
+            'phone_number_full': fake.phone_number,
+            'license_plate': self.license_plate
+        }
+
+        if data_type not in func_lookup:
+            raise ValueError("Data type must be one of " + str(list(func_lookup.keys())))
+
+        datagen_func = func_lookup[data_type]
+        return pd.Series((datagen_func() for _ in range(num)))
+
+    def _validate_args(self, num, fields):

```

Comment:
Printing an error and returning `None` can be unintuitive in non-interactive scenarios. Raising an exception will make argument validation very explicit.